guides/source/documents.yaml

```diff
@@ -72,6 +72,10 @@
       url: active_support_core_extensions.html
       description: This guide documents the Ruby core extensions defined in Active Support.
     -
+      name: Active Support On Load Hooks
```

Comment:
Agree, too narrow for a dedicated guide.

The initialization guide is basically abandoned though. It was started some time ago, but it has a code walkthrough approach that I do not like, because I'd prefer that guide to be the bird's eye view of what runs when a Rails developer need to know. Would be closer to what I covered in [this talk](https://www.youtube.com/watch?v=vSza2FZucV8).

I believe the main use case from a developer point of view is railties/engines, @rafaelfranca what do you think about adding that to the engines guides meanwhile?

I believe the motivation in the proposed docs is a bit off also. It says "you could load classes from the framework, but would be costly, so you have this trick to lazy load". I think it would better say: "Rails is responsible for what is loaded when, you should not load frameworks by yourself because that violates the contract, on_load is the API to integrate with the Rails boot process".